### PR TITLE
Add option to always record telemetry during token fetch

### PR DESCRIPTION
**Pull request by jatgarg**
Add option to always record telemetry during token fetch during trees latest as this helps hosts to track the token fetch time.

### CODE REVIEW

**Comment by jatgarg** [comment]
@zlatkovsky fyi

File: packages/drivers/odsp-driver/src/fetchSnapshot.ts [hunk 1]
```diff
@@ -176,7 +179,7 @@ async function fetchLatestSnapshotCore(
                 errorType: "access denied",
             }, tokenFetchOptions.previousError);
         }
-        const storageToken = await storageTokenFetcher(tokenFetchOptions, "TreesLatest");
+        const storageToken = await storageTokenFetcher(tokenFetchOptions, "TreesLatest", true);
```
Comment (Zlatkovsky) [inline packages/drivers/odsp-driver/src/fetchSnapshot.ts:182]: Do you guys do the convention of adding a comment for the param name? I.e.,:
`true /* alwaysRecordTokenFetchTelemetry */`

Comment (jatgarg) [reply packages/drivers/odsp-driver/src/fetchSnapshot.ts:182]: No

File: packages/drivers/odsp-driver/src/odspUtils.ts [hunk 2]
```diff
@@ -272,8 +272,9 @@ export function toInstrumentedOdspTokenFetcher(
                 // This event alone generates so many events that is materially impacts cost of telemetry
                 // Thus do not report end event when it comes back quickly.
                 // Note that most of the hosts do not report if result is comming from cache or not,
-                // so we can't rely on that here
-                if (event.duration >= 32) {
+                // so we can't rely on that here. But always record if specified explicitly for cases such as
+                // calling trees/latest during load.
+                if (event.duration >= 32 || alwaysRecordTokenFetchTelemetry) {
```
Comment (Zlatkovsky) [inline packages/drivers/odsp-driver/src/odspUtils.ts:277]: Super nit, I would reverse the order such that it's "if want to always record telemetry *or* duration is >= 32"

File: packages/drivers/odsp-driver/src/fetchSnapshot.ts [hunk 3]
```diff
@@ -152,7 +154,8 @@ async function redeemSharingLink(
 
 async function fetchLatestSnapshotCore(
     odspResolvedUrl: IOdspResolvedUrl,
-    storageTokenFetcher: (options: TokenFetchOptions, name: string) => Promise<string | null>,
+    storageTokenFetcher: (options: TokenFetchOptions, name: string, alwaysRecordTokenFetchTelemetry?: boolean) =>
```
Comment (Zlatkovsky) [inline packages/drivers/odsp-driver/src/fetchSnapshot.ts:157]: It seems like there's a bunch of these
```
 storageTokenFetcher: (options: TokenFetchOptions, name: string, alwaysRecordTokenFetchTelemetry?: boolean) =>
        Promise<string | null>
```
lines.

Would it make sense to create a type and use it throughout?

```
type StorageTokenFetcherCallback = (options: TokenFetchOptions, name: string, alwaysRecordTokenFetchTelemetry?: boolean) =>
    Promise<string | null>;

<...>

storageTokenFetcher: StorageTokenFetcherCallback

**Review by Zlatkovsky — APPROVED** [review]

**Review by Zlatkovsky — APPROVED** [review]

File: packages/drivers/odsp-driver/src/fetchSnapshot.ts (hunk 1 above)
Comment (vladsud) [reply packages/drivers/odsp-driver/src/fetchSnapshot.ts:179]: I do and prefer it :)

**Comment by vladsud** [comment]
I'm curious what this change actually solves? I.e. what is the problem and why it's a solution to this problem?
I'm not sure it gives us anything.

**Review by anthony-murphy — APPROVED** [review]

**Comment by jatgarg** [comment]
> I'm curious what this change actually solves? I.e. what is the problem and why it's a solution to this problem?
> I'm not sure it gives us anything.

Actually @Zlatkovsky wants to analyze the sessions and look at the token fetch telemetry. And we don't always record it as per the condition. So here I made it opt in and always record it for the trees latest token fetch call.

**Pull request merged**